src/tools/cephfs/top/cephfs-top

```diff
@@ -517,70 +591,87 @@ class FSTop(object):
                 return
 
-    def refresh_clients(self, x_coord_map, stats_json):
+    def refresh_clients(self, x_coord_map, stats_json, fs_name):
         counters = [m.upper() for m in stats_json[GLOBAL_COUNTERS_KEY]]
         y_coord = 0
-        for client_id, metrics in stats_json[GLOBAL_METRICS_KEY].items():
-            self.refresh_client(client_id,
-                                metrics,
-                                counters,
-                                stats_json[CLIENT_METADATA_KEY][client_id],
-                                x_coord_map,
-                                y_coord)
-            y_coord += 1
-
-    def refresh_main_window(self, x_coord_map, stats_json):
+        hlen = min(self.width - 2, len(FS_TOP_NAME_TOPL_FMT))
+        res = stats_json[CLIENT_METADATA_KEY].get(fs_name)
+        if res is None:
+            client_cnt = 0
+            self.topl.addnstr(0, 0, FS_TOP_NAME_TOPL_FMT.format(
+                fs_name=fs_name, client_count=client_cnt), hlen)
+            self.topl.refresh()
+        else:
+            client_cnt = len(res)
+            self.topl.addnstr(0, 0, FS_TOP_NAME_TOPL_FMT.format(
+                fs_name=fs_name, client_count=client_cnt), hlen)
+            self.topl.refresh()
+            for client_id, metrics in \
+                    stats_json[GLOBAL_METRICS_KEY][fs_name].items():
+                self.refresh_client(
+                    client_id, metrics, counters,
+                    stats_json[CLIENT_METADATA_KEY][fs_name][client_id],
+                    x_coord_map, y_coord)
+                y_coord += 1
+
+    def refresh_main_window(self, x_coord_map, stats_json, fs_name):
         if self.mainw is None:
             return
-        self.refresh_clients(x_coord_map, stats_json)
+        self.refresh_clients(x_coord_map, stats_json, fs_name)
         self.mainw.refresh()
 
-    def refresh_header(self, stats_json):
+    def refresh_header(self, stats_json, headline):
         hlen = self.width - 2
+        num_clients, num_mounts, num_kclients, num_libs = 0, 0, 0, 0
         if not stats_json['version'] == FS_TOP_SUPPORTED_VER:
             self.header.addnstr(0, 0, 'perf stats version mismatch!', hlen)
             return False
-        client_metadata = stats_json[CLIENT_METADATA_KEY]
-        num_clients = len(client_metadata)
-        num_mounts = len([client for client, metadata in client_metadata.items() if
-                          CLIENT_METADATA_MOUNT_POINT_KEY in metadata
-                          and metadata[CLIENT_METADATA_MOUNT_POINT_KEY] != 'N/A'])
-        num_kclients = len([client for client, metadata in client_metadata.items() if
-                            "kernel_version" in metadata])
-        num_libs = num_clients - (num_mounts + num_kclients)
+        for fs_name in fs_list:
+            if stats_json[CLIENT_METADATA_KEY].get(fs_name) is not None:
+                client_metadata = stats_json[CLIENT_METADATA_KEY].get(fs_name)
```

Comment:
You are `get` ing the same thing 2 times. Assign to `client_metadata` first. Use default value wherever possible.

Reply:
You could use a default empty dict here also. So it would be simple as `if client_metadata:`.